Apps/Database/src/Migrations/20220518183026_UserProfileToSFK.cs

```diff
@@ -0,0 +1,97 @@
+﻿// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+#pragma warning disable CS1591
+// <auto-generated />
+using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace HealthGateway.Database.Migrations
+{
+    public partial class UserProfileToSFK : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            string schema = "gateway";
+            migrationBuilder.Sql(@$"DROP TRIGGER IF EXISTS ""UserProfileHistoryLoginTrigger"" ON {schema}.""UserProfile"";");
+
+            migrationBuilder.AddColumn<Guid>(
+                name: "TermsOfServiceId",
+                schema: "gateway",
+                table: "UserProfileHistory",
+                type: "uuid",
+                nullable: true);
+
+            migrationBuilder.AddColumn<Guid>(
+                name: "TermsOfServiceId",
+                schema: "gateway",
+                table: "UserProfile",
+                type: "uuid",
+                nullable: false,
+                defaultValue: new Guid("c99fd839-b4a2-40f9-b103-529efccd0dcd"));
+
+            migrationBuilder.CreateIndex(
+                name: "IX_UserProfile_TermsOfServiceId",
+                schema: "gateway",
+                table: "UserProfile",
+                column: "TermsOfServiceId");
+
+            migrationBuilder.AddForeignKey(
+                name: "FK_UserProfile_LegalAgreement_TermsOfServiceId",
+                schema: "gateway",
+                table: "UserProfile",
+                column: "TermsOfServiceId",
+                principalSchema: "gateway",
+                principalTable: "LegalAgreement",
+                principalColumn: "LegalAgreementsId",
+                onDelete: ReferentialAction.Cascade);
```

Comment:
Instead of Cascade, wouldn't Restrict make more sense here?

Reply:
Nice catch!  Updated